jina/executors/dump.py

```diff
@@ -0,0 +1,132 @@
+import os
+import sys
+from typing import Iterable, Tuple
+
+import numpy as np
+
+from jina.enums import BetterEnum
+
+
+class DumpTypes(BetterEnum):
+    """The enum of dump formats"""
+
+    DEFAULT = 0
+    NUMPY = 1
+
+
+BYTE_PADDING = 4
+
+
+class DumpPersistor:
+    """
+    Class for creating and importing from dumps
+
+    Do NOT instantiate. Only provides static methods
+    """
+
+    @staticmethod
+    def export_dump_streaming(
+        path: str,
+        shards: int,
+        size: int,
+        data: Iterable[Tuple[str, np.array, bytes]],
+    ):
+        """Export the data to a path, based on sharding,
+
+        :param path: path to dump
+        :param shards: the nr of shards this pea is part of
+        :param size: total amount of entries
+        :param data: the generator of the data (ids, vectors, metadata)
+        """
+        if os.path.exists(path):
+            raise Exception(f'path for dump {path} already exists. Not dumping...')
+        size_per_shard = size // shards
+        extra = size % shards
+        shard_range = list(range(shards))
+        for shard_id in shard_range:
+            if shard_id == shard_range[-1]:
+                size_this_shard = size_per_shard + extra
+            else:
+                size_this_shard = size_per_shard
+            shard_path = os.path.join(path, str(shard_id))
+            shard_written = 0
+            os.makedirs(shard_path)
+            vectors_fh, metas_fh, ids_fh = DumpPersistor._get_file_handlers(shard_path)
```

Comment:
these file handlers should be context managed

Reply:
That would make this method even messier. Not sure it's worth the mess